types/react/index.d.ts

```diff
@@ -50,7 +50,7 @@ declare namespace React {
     // ----------------------------------------------------------------------
 
     type ReactType<P = any> = string | ComponentType<P>;
-    type ComponentType<P = {}> = ComponentClass<P> | StatelessComponent<P>;
+    type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
```

Comment:
I left it out of the scope of this PR, but what are people's thoughts on changing `ComponentClass` to `ClassComponent` to make it consistent with `FunctionComponent`?

Reply:
I'm... ambivalent about that. On one hand, I kinda like how `ComponentClass` sounds more, but on the other hand, it's probably a bias towards what I'm already used to.

Reply:
I mostly base this on the React docs, which use the phrase "class component" all over the place, but yeah I think it's something that definitely warrants more discussion outside this PR.